dandi/dandiapi.py

```diff
@@ -318,10 +318,16 @@ def __init__(
         self, api_url: Optional[str] = None, token: Optional[str] = None
     ) -> None:
         """
-        Construct a client instance from a server's base API URL and an
-        optional authentication token/API key.  If no URL is supplied, the URL
-        of the instance named in the :envvar:`DANDI_INSTANCE` environment
-        variable (default value: ``dandi``) is used.
+        Construct a client instance.
+
+        :param api_url: API URL. For DANDI production, use ``"dandi"``. For DANDI staging,
+        use ``"https://api-staging.dandiarchive.org/api"``. If no URL is supplied, the
+        value of the  :envvar:`DANDI_INSTANCE` environment variable is used. If that
+        variable is not defined, ``"dandi"`` is used by default.
+        :type method: str
+        :param token: User API Key. By default, uses :envvar:`DANDI_API_KEY`.
```

Comment:
"By default, uses :envvar:`DANDI_API_KEY`." — Incorrect.  By default, no key is used.

Reply:
It's not that different from the command line, as the command line only authenticates if there's a need to (i.e., if downloading an embargoed Dandiset or if uploading anything).  There are circumstances in which one would want to access the API without authenticating (e.g., in order to see only what's publicly available), and I believe it's better for a low-level constructor to default to not pulling things from the environment.  (I'd also prefer not honoring `DANDI_INSTANCE` here, but Yarik asked for it.)

Reply:
ok, fair enough